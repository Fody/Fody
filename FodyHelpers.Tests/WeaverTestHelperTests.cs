using System;
using System.IO;
using Fody;
using Mono.Cecil;
using Xunit;
using Xunit.Abstractions;

public class WeaverTestHelperTests :
    XunitApprovalBase
{
    [Fact]
    public void Run()
    {
        var weaver = new TargetWeaver();
        var result = weaver.ExecuteTestRun("DummyAssembly.dll");
        Verify(result);
    }

    static void Verify(TestResult result)
    {
        ObjectApprover.Verify(new
            {
                result.Errors,
                result.Messages,
                result.Warnings,
                result.AssemblyPath,
                result.Assembly.FullName
            },
            ScrubCurrentDirectory);
    }

    static string ScrubCurrentDirectory(string s)
    {
        return s.Replace(@"\\", @"\").Replace(Environment.CurrentDirectory, "");
    }

    [Fact]
    public void WithCustomAssemblyName()
    {
        var assemblyPath = Path.Combine(Environment.CurrentDirectory, "DummyAssembly.dll");
        var weaver = new TargetWeaver();
        var result = weaver.ExecuteTestRun(
            assemblyPath: assemblyPath,
            assemblyName: "NewName");
        Verify(result);
    }

    [Fact]
    public void WeaverUsingSymbols()
    {
        var assemblyPath = Path.Combine(Environment.CurrentDirectory, "DummyAssembly.dll");
        var weaver = new WeaverUsingSymbols();
        var result = weaver.ExecuteTestRun(assemblyPath, writeSymbols:true);
        var module = ModuleDefinition.ReadModule(assemblyPath,new ReaderParameters {ReadSymbols = true});
        Assert.True(module.HasSymbols);

        Verify(result);
    }

    public WeaverTestHelperTests(ITestOutputHelper output) :
        base(output)
    {
    }
}