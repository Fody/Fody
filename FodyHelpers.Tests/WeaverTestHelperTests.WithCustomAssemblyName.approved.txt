{
  Messages: [
    {
      Text: '\tRemoving reference to \'Tests\'.',
      MessageImportance: 'Low'
    }
  ],
  AssemblyPath: '\fodytemp\NewName.dll',
  FullName: 'NewName, Version=0.0.0.0, Culture=neutral, PublicKeyToken=1ca091877d12ca03'
}